File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/PolarDiskSpaceData.kt
package com.polar.sdk.api.model

import protocol.PftpResponse

/**
 * Disk space data in bytes.
 */
data class PolarDiskSpaceData(
    val totalSpace: Long,
    val freeSpace: Long
) {
    companion object {
        fun fromProto(proto: PftpResponse.PbPFtpDiskSpaceResult): PolarDiskSpaceData {
            return PolarDiskSpaceData(
                totalSpace = proto.fragmentSize * proto.totalFragments,
                freeSpace = proto.fragmentSize * proto.freeFragments
            )
        }
    }
}

File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/model/PolarDiskSpaceData.swift
//  Copyright © 2023 Polar. All rights reserved.

public struct PolarDiskSpaceData {
    public let totalSpace: UInt64
    public let freeSpace: UInt64
    
    static func fromProto(proto: Protocol_PbPFtpDiskSpaceResult) -> PolarDiskSpaceData {
        return PolarDiskSpaceData(
            totalSpace: UInt64(proto.fragmentSize) * proto.totalFragments,
            freeSpace: UInt64(proto.fragmentSize) * proto.freeFragments
        )
    }
}
